### PR TITLE
Add roles

### PR DESCRIPTION
**Pull request by kont-noor**
### Changes:
- Add `Role` model
- Add `AccountsRole` relation
- Add `Account#has_role?` method
- Add specs

### CODE REVIEW

File: app/models/incarnator/account.rb [hunk 1]
```diff
@@ -10,5 +13,9 @@ class Account < ActiveRecord::Base
            :confirmable
 
     validates :email, :user_name, presence: true, uniqueness: true
+
+    def has_role?(role_name)
+      roles.find_by(name: role_name).present?
```
Comment (DevMike) [inline app/models/incarnator/account.rb]: is 'present?' required here?

File: db/migrate/20151124211951_create_incarnator_accounts_roles.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class CreateIncarnatorAccountsRoles < ActiveRecord::Migration
+  def change
+    create_table :incarnator_accounts_roles do |t|
+      t.references :account, null: false
+      t.references :role,    null: false
+
+      t.timestamps null: false
+    end
+
+    add_index :incarnator_accounts_roles, :account_id
```
Comment (DevMike) [inline db/migrate/20151124211951_create_incarnator_accounts_roles.rb]: don't see sense for separated indexes on both account_id and user_id, the unique index is sufficient as for me

File: spec/models/incarnator/accounts_role_spec.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+module Incarnator
+  RSpec.describe AccountsRole, type: :model do
+    it { is_expected.to belong_to(:account) }
+    it { is_expected.to belong_to(:role) }
+    it { is_expected.to validate_presence_of(:account) }
```
Comment (DevMike) [inline spec/models/incarnator/accounts_role_spec.rb]: pls separate validations with empty lines like there https://github.com/kont-noor/Incarnator/pull/2/files#diff-7d1c9612982cf6c395197bfb75c78babR7

File: spec/models/incarnator/account_spec.rb [hunk 4]
```diff
@@ -2,9 +2,27 @@
...
     it { is_expected.to validate_uniqueness_of(:email) }
+
+    describe "#has_role?" do
+      subject { FactoryGirl.create(:account) }
+      let(:role) { FactoryGirl.create(:incarnator_role) }
+
+      it "returns true if account has role" do
+        subject.roles << role
+
+        expect(subject.has_role?(role.name)).to eq(true)
```
Comment (DevMike) [inline spec/models/incarnator/account_spec.rb:20]: might be be_false/be_true

Comment (kont-noor) [inline spec/models/incarnator/account_spec.rb:20]: there is no method `#true?` or `#false` for `Object`

File: app/models/incarnator/account.rb (hunk 1 above)
Comment (skarm61) [inline app/models/incarnator/account.rb]: `roles.where(name: role_name).any?`
it doesn't take any data from db. It just execute sql with 'COUNT'

Comment (kont-noor) [inline app/models/incarnator/account.rb]: @skarm61 it makes sense

Comment (DevMike) [inline app/models/incarnator/account.rb]: exists? works faster

**Pull request closed without merging**